### PR TITLE
[MIG] sales_team to new API

### PR DESCRIPTION
**Pull request by lla-odoo**
https://www.odoo.com/web#id=13726&view_type=form&model=project.task&menu_id=3942&action=327

### CODE REVIEW

File: addons/sales_team/data/sales_team_demo.xml [hunk 1]
```diff
@@ -8,7 +8,7 @@
         <record id="team_sales_department" model="crm.team">
             <field name="member_ids" eval="[(4, ref('base.user_demo'))]"/>
         </record>
-        <record model="crm.team" id="crm_team_1">
+        <record id="crm_team_1" model="crm.team">
```
Comment (tde-banana-odoo) [inline addons/sales_team/data/sales_team_demo.xml]: Small remark: unnecessary change.

File: addons/sales_team/models/crm_team.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class CrmTeam(models.Model):
+    _name = "crm.team"
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+    _description = "Sales Team"
+    _order = "name"
+
+    @api.model
+    def _get_default_team_id(self, user_id=None):
```
Comment (tde-banana-odoo) [inline addons/sales_team/models/crm_team.py]: Actually I think you have to use a return decorator. The method should return an id in the old API, but a record set in the new API, if I am not wrong.

File: addons/sales_team/models/crm_team.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models
```
Comment (tde-banana-odoo) [inline addons/sales_team/models/crm_team.py]: You can use from odoo import ... It should work in master.

File: addons/sales_team/models/crm_team.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            user_id = self.env.uid
+
+        team_id = self.env.context.get('default_team_id')
+        if not team_id:
+            team = self.search(['|', ('user_id', '=', user_id), ('member_ids', 'in', [user_id])], limit=1)
+            team_id = team.id if team else self.env.ref('sales_team.team_sales_department').id
+        return team_id
+
+    name = fields.Char('Sales Team', size=64, required=True, translate=True)
+    code = fields.Char('Code', size=8)
```
Comment (tde-banana-odoo) [inline addons/sales_team/models/crm_team.py]: Side question: why limiting the size of the code field ? Can't I use the "ILoveBananasAndPumpkins" code ?

File: addons/sales_team/models/crm_team.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        if user_id is None:
+            user_id = self.env.uid
+
+        team_id = self.env.context.get('default_team_id')
+        if not team_id:
+            team = self.search(['|', ('user_id', '=', user_id), ('member_ids', 'in', [user_id])], limit=1)
+            team_id = team.id if team else self.env.ref('sales_team.team_sales_department').id
+        return team_id
+
+    name = fields.Char('Sales Team', size=64, required=True, translate=True)
```
Comment (tde-banana-odoo) [inline addons/sales_team/models/crm_team.py]: Side question: is it really necessary to limit the size of the sales team name ?

File: addons/sales_team/models/crm_team.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    code = fields.Char('Code', size=8)
+    active = fields.Boolean(string='Active', help="If the active field is set to true, it will allow you to hide the sales team without removing it.", default=True)
```
Comment (tde-banana-odoo) [inline addons/sales_team/models/crm_team.py]: Help = not(behavior) if I am not wrong.

File: addons/sales_team/models/crm_team.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env['res.company']._company_default_get('crm.team'))
+    user_id = fields.Many2one('res.users', string='Team Leader')
+    member_ids = fields.One2many('res.users', 'sale_team_id', string='Team Members')
+    reply_to = fields.Char(string='Reply-To', size=64, help="The email address put in the 'Reply-To' of all emails sent by Odoo about cases in this sales team")
```
Comment (tde-banana-odoo) [inline addons/sales_team/models/crm_team.py]: Side question: is the size necessary ?

File: addons/sales_team/models/crm_team.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    working_hours = fields.Float(string='Working Hours', digits=(16, 2))
```
Comment (tde-banana-odoo) [inline addons/sales_team/models/crm_team.py]: Side question is this field used somewhere ? Any idea of its use ?

File: addons/sales_team/models/crm_team.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    color = fields.Integer(string='Color Index')
```
Comment (tde-banana-odoo) [inline addons/sales_team/models/crm_team.py]: While we are at it, an help message could be great (like the color range, things like that).

File: addons/sales_team/models/crm_team.py (hunk 8 above)
Comment (lla-odoo) [inline addons/sales_team/models/crm_team.py]: I checked in children models, but none of them make use of it. I removed it

File: addons/sales_team/models/crm_team.py (hunk 6 above)
Comment (lla-odoo) [inline addons/sales_team/models/crm_team.py]: Actually it describes the FALSE behavior...

File: addons/sales_team/models/crm_team.py (hunk 5 above)
Comment (lla-odoo) [inline addons/sales_team/models/crm_team.py]: is this perhaps for a graphical reason?

File: addons/sales_team/data/sales_team_data.xml [hunk 10]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
+<odoo>
     <data noupdate="1">
-        <record model="crm.team" id="team_sales_department">
+        <record id="team_sales_department" model="crm.team">
```
Comment (tde-banana-odoo) [inline addons/sales_team/data/sales_team_data.xml]: Small remark: unnecessary change.

File: addons/sales_team/security/sales_team_security.xml [hunk 11]
```diff
@@ -1,10 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
+<odoo>
     <data noupdate="0">
-        <record model="ir.rule" id="sale_team_comp_rule">
+        <record id="sale_team_comp_rule" model="ir.rule">
```
Comment (tde-banana-odoo) [inline addons/sales_team/security/sales_team_security.xml]: Small remark: unnecessary change
